viewer/packages/rendering/src/GeometryBatchingManager.ts

```diff
@@ -201,6 +219,42 @@ export class GeometryBatchingManager {
     return interleavedBufferView;
   }
 
+  private getTreeIndexAttribute(
+    instanceAttributes: { name: string; attribute: THREE.InterleavedBufferAttribute }[]
+  ): THREE.InterleavedBufferAttribute {
+    let treeIndexAttribute: THREE.InterleavedBufferAttribute | undefined = undefined;
+    for (let i = 0; i < instanceAttributes.length; i++) {
+      if (instanceAttributes[i].name == 'a_treeIndex') {
+        treeIndexAttribute = instanceAttributes[i].attribute;
+      }
+    }
+
+    assert(treeIndexAttribute !== undefined);
+
+    return treeIndexAttribute;
+  }
+
+  private removeTreeIndicesFromMeshUserData(
+    mesh: THREE.Mesh,
+    updateRange: { byteOffset: number; byteCount: number }
+  ): void {
+    const { byteOffset, byteCount } = updateRange;
+
+    const interleavedAttributes = this.getAttributes(mesh.geometry, THREE.InterleavedBufferAttribute);
+
+    const treeIndexAttribute = this.getTreeIndexAttribute(interleavedAttributes);
+    const typedArray = treeIndexAttribute.data.array as TypedArray;
+    const instanceByteSize = treeIndexAttribute.data.stride * typedArray.BYTES_PER_ELEMENT;
+
+    const firstWholeInstanceIndex = Math.ceil(byteOffset / instanceByteSize);
```

Comment:
Does it make sense to math.ceil here? I would think that if byteOffset is not divisible by instanceByteSize then the state of the buffer is invalid.

Reply:
I agree - putting in a couple of asserts instead.